torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -324,17 +324,23 @@ def apply_fsdp(
         )
     fully_shard(model, **fsdp_config, reshard_after_forward=not pp_enabled)
 
-    if pp_enabled:
-        # TODO
-        # This PR https://github.com/pytorch/pytorch/pull/129519 added a safety check to avoid using 2D/3D DCP since
-        # without strided sharding, DCP can not safely support resharding for 2D/3D.  However, for PP to work, even
-        # without resharding, we load a seed-checkpoint and need to disable the safety mechanism.  This hack should be
-        # removed after strided sharding is landed in DCP.
-        for module in model.modules():
-            assert len(module._load_state_dict_pre_hooks) <= 1
+    # TODO: This PR https://github.com/pytorch/pytorch/pull/129519 added a safety check
+    # to avoid using 2D/3D DCP since without strided sharding, DCP can not safely support
+    # resharding for 2D/3D. We keep this safety check disablement here for now and will
+    # remove it when PyTorch gets the next minor release (PyTorch 2.5).
```

Comment:
We can safely check `torch.__version__` to make sure that only the nightly after a certain date to enable 2D/3D DCP state dict

Reply:
I guess we can check if the nightly version is >= your commit (https://github.com/pytorch/pytorch/pull/131408), we can remove the hooks. Otherwise, we should keep it.

Reply:
@wanchaol that would stop users who use build-from-source pytorch from enabling 2D/3D DCP. Is there a way to examine if a `torch.__version__` (e.g. `2.5.0a0+git9cca049`) is newer than a given commit hash? cc @wz337

Reply:
I think there's no obvious way to know whether the current commit hash is newer than a certain hash, IMO we should just check the nightly version instead (skip the build-from-source check)